Fix Sporadic Test Failures Due to Expired Redis Cache

### Bug Description

[intermittent failing test](https://github.com/ethyca/fidesops/actions/runs/3091373106/attempts/1)


Troubleshoot sporadic failures.  Data is not in the redis cache even though we've set it just prior to running the privacy request.

```bash
_____ TestPrivacyRequestsManualWebhooks.test_pass_on_empty_confirmed_input _____

self = <tests.ops.service.privacy_request.request_runner_service_test.TestPrivacyRequestsManualWebhooks object at 0x7fc4b4172130>
mock_upload = <MagicMock name='upload' id='140482412235600'>
integration_manual_webhook_config = <fidesops.ops.models.connectionconfig.ConnectionConfig object at 0x7fc49c467bb0>
access_manual_webhook = <fidesops.ops.models.manual_webhook.AccessManualWebhook object at 0x7fc49c467af0>
policy = <fidesops.ops.models.policy.Policy object at 0x7fc49c45efa0>
run_privacy_request_task = <@task: fidesops.ops.service.privacy_request.request_runner_service.run_privacy_request of celery.tests at 0x7fc4afc789d0>
privacy_request_requires_input = <fidesops.ops.models.privacy_request.PrivacyRequest object at 0x7fc49c474a30>
db = <sqlalchemy.orm.session.ExtendedSession object at 0x7fc4b41b78b0>

    @mock.patch("fidesops.ops.service.privacy_request.request_runner_service.upload")
    def test_pass_on_empty_confirmed_input(
        self,
        mock_upload,
        integration_manual_webhook_config,
        access_manual_webhook,
        policy,
        run_privacy_request_task,
        privacy_request_requires_input: PrivacyRequest,
        db,
    ):
        privacy_request_requires_input.cache_manual_webhook_input(
            access_manual_webhook,
            {},
        )
    
        run_privacy_request_task.delay(privacy_request_requires_input.id).get(
            timeout=PRIVACY_REQUEST_TASK_TIMEOUT
        )
    
        db.refresh(privacy_request_requires_input)
>       assert privacy_request_requires_input.status == PrivacyRequestStatus.complete
E       AssertionError: assert <PrivacyReque...quires_input'> == <PrivacyReque...e: 'complete'>
E         - complete
E         + requires_input
```

#### Steps to Reproduce

1. _Do this..._
1. _Then This..._

### Expected behavior

_A description of what you expected to happen._

### Screenshots

_If applicable, add screenshots to help explain your problem._

### Environment

* Version:
* OS:
* Python Version:
* Docker Version:

### Additional context

_Add any other context about the problem here._

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -19,6 +19,11 @@ The types of changes are:
 
 ## [Unreleased](https://github.com/ethyca/fidesops/compare/1.7.2...main)
 
+
+### Fixed
+* Distinguish whether webhook has been visited and no fields were found, versus never visited [#1339](https://github.com/ethyca/fidesops/pull/1339)
+* Fix Redis Cache Early Expiration in Tests [#1358](https://github.com/ethyca/fidesops/pull/1358)
+
 ## [1.8.0](https://github.com/ethyca/fidesops/compare/1.8.0...main)
 
 ### Developer Experience
@@ -90,7 +95,6 @@ The types of changes are:
 * Remove masking of redis error log [#1288](https://github.com/ethyca/fidesops/pull/1288)
 * Logout with malformed or expired token [#1305](https://github.com/ethyca/fidesops/pull/1305)
 * The `toml` package is now included in the list of direct dependencies (`requirements.txt`) [#1338](https://github.com/ethyca/fidesops/pull/1338)
-* Distinguish whether webhook has been visited and no fields were found, versus never visited [#1339](https://github.com/ethyca/fidesops/pull/1339)
 * Fix bug where erasure counts instead of access results may be retrieved for certain collections on request retry [#1349](https://github.com/ethyca/fidesops/pull/1349)
 
 ### Security

--- a/tests/ops/fixtures/application_fixtures.py
+++ b/tests/ops/fixtures/application_fixtures.py
@@ -17,7 +17,7 @@ from sqlalchemy.orm import Session
 from sqlalchemy.orm.exc import ObjectDeletedError
 
 from fidesops.ops.api.v1.scope_registry import PRIVACY_REQUEST_READ, SCOPE_REGISTRY
-from fidesops.ops.core.config import config
+from fidesops.ops.core.config import FidesopsConfig, config
 from fidesops.ops.models.connectionconfig import (
     AccessLevel,
     ConnectionConfig,
@@ -1257,3 +1257,13 @@ def application_user(
     oauth_client.save(db=db)
     yield user
     user.delete(db=db)
+
+
+@pytest.fixture(scope="function")
+def short_redis_cache_expiration() -> FidesopsConfig:
+    original_value: int = config.redis.default_ttl_seconds
+    config.redis.default_ttl_seconds = (
+        1  # Set redis cache to expire very quickly for testing purposes
+    )
+    yield config
+    config.redis.default_ttl_seconds = original_value

--- a/tests/ops/service/privacy_request/request_runner_service_test.py
+++ b/tests/ops/service/privacy_request/request_runner_service_test.py
@@ -13,8 +13,6 @@ from sqlalchemy.orm import Session
 
 from fidesops.ops.common_exceptions import (
     ClientUnsuccessfulException,
-    EmailDispatchException,
-    IdentityNotFoundException,
     PrivacyRequestPaused,
 )
 from fidesops.ops.core.config import config
@@ -1472,10 +1470,8 @@ class TestRunPrivacyRequestRunsWebhooks:
         db,
         privacy_request,
         policy_pre_execution_webhooks,
+        short_redis_cache_expiration,  # Fixture forces cache to expire quickly
     ):
-        config.redis.default_ttl_seconds = (
-            1  # Set redis cache to expire very quickly for testing purposes
-        )
         mock_trigger_policy_webhook.side_effect = PrivacyRequestPaused(
             "Request received to halt"
         )

